src/ray/core_worker/object_interface.cc

```diff
@@ -1,17 +1,20 @@
 #include "ray/core_worker/object_interface.h"
 #include "ray/common/ray_config.h"
+#include "ray/core_worker/store_provider/local_plasma_provider.h"
 #include "ray/core_worker/store_provider/plasma_store_provider.h"
 
 namespace ray {
 
 CoreWorkerObjectInterface::CoreWorkerObjectInterface(
     WorkerContext &worker_context, std::unique_ptr<RayletClient> &raylet_client,
     const std::string &store_socket)
-    : worker_context_(worker_context), raylet_client_(raylet_client) {
-  store_providers_.emplace(
-      static_cast<int>(StoreProviderType::PLASMA),
-      std::unique_ptr<CoreWorkerStoreProvider>(
-          new CoreWorkerPlasmaStoreProvider(store_socket, raylet_client_)));
+    : worker_context_(worker_context),
+      raylet_client_(raylet_client),
+      store_socket_(store_socket) {
+  store_providers_.emplace(static_cast<int>(StoreProviderType::LOCAL_PLASMA),
+                           CreateStoreProvider(StoreProviderType::LOCAL_PLASMA));
+  store_providers_.emplace(static_cast<int>(StoreProviderType::PLASMA),
+                           CreateStoreProvider(StoreProviderType::PLASMA));
```

Comment:
Suggest changing these lines to `AddStoreProvider(StoreProviderType::LOCAL_PLASMA); AddStoreProvider(StoreProviderType::PLASMA);`. Then inside `AddStoreProvider`, we put the newly created instance into `store_providers_` map. This will make it more clear.

Reply:
yes thanks for the suggestion.